src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -632,15 +679,33 @@ ConvSolution InitInvokerFactoryNCHW(const ExecutionContext& ctx,
                                                    tr_ptrs[0]->GetBufferPtr(),
                                                    tr_ptrs[1]->GetBufferPtr(),
                                                    tr_ptrs[2]->GetBufferPtr());
-            float conv_time   = 0;
-            conv_time += invoker_ptr->Run(argument_ptr.get(),
-                                          {handle.GetStream(), handle.IsProfilingEnabled()});
-
-            if(handle.IsProfilingEnabled())
+            if constexpr(std::is_same<CastType, miopen::conv::WrWInvokeParams>::value)
             {
-                handle.AccumKernelTime(conv_time);
+#if MIOPEN_BACKEND_HIP
+                HipEventPtr start = nullptr;
+                HipEventPtr stop  = nullptr;
```

Comment:
can you explain why we need to add extra HipEventPtr variable `start`/`stop` considering we have these hipevent inside handler?
This PR looks like only move the hipevent and profiling related stuff out of handler, and how could it solve the precision issue of wrw kernel?

Reply:
@carlushuang  In the currently implementation, we use the profiling time that get directly from ck-api (ck invoker run fuction). When the ck-profiling is enabled, the ck wrw kernel introduces the errors that cause the precision issue. The ck-profiling should be set to false to avoid the precision issue but the time will be 0 in this case. I tried to utilize the handler to get the time. However, the current timer scheme in handler does not work properly, including approach 1 in this [comment](https://github.com/ROCm/MIOpen/pull/2429#discussion_r1360795493) Artem mentioned. 
So I proposed this workaround that using hipevent to get the correct time for the ck kernels.

Reply:
@iq136boy Thanks for the explanation. I'll provide next review soon.